feat: implement refocusing trigger-element after closing modal

RafikiTiki opened the pull request:
In a project which I'm part of right now we're working towards compatibility with WCAG spec. When using current implementation of the `<Modal />` component I've seen that you've handled focus trap problem in very clean manner but there's one more thing that I believe that should be implemented to be even more compliant with the spec.

> A Web page implements modal dialogs via scripting. When the trigger button is activated, a dialog opens and focus is set to the first interactive element in the dialog. As long as the dialog is open, focus is limited to the elements of the dialog. **When the dialog is dismissed, focus returns to the button or the element following the button.**

source: https://www.w3.org/WAI/WCAG21/Understanding/focus-order Examples point 3

Right now after closing modal, focus is set to the `<body>` element. User relaying solely on keyboard for navigation now needs to _tab_ his/hers way to the button once more which is frustrating and definitely not a good UX.

This PR is implementing _refocus element that triggered modal after closing it_ mechanism.

necolas commented on packages/react-native-web/src/exports/Modal/ModalFocusTrap.js:
```diff
@@ -123,6 +124,19 @@ const ModalFocusTrap = ({ active, children }: ModalFocusTrapProps) => {
+    if (canUseDOM) {
+      lastFocusedElementOutsideTrap.current = document.activeElement;
+      return () => {
+        if (lastFocusedElementOutsideTrap.current) {
```
This value needs to be stored in the closure rather than read in the cleanup callback

necolas commented on packages/react-native-web/src/exports/Modal/ModalFocusTrap.js:
```diff
@@ -123,6 +124,19 @@ const ModalFocusTrap = ({ active, children }: ModalFocusTrapProps) => {
+      lastFocusedElementOutsideTrap.current = document.activeElement;
+      return () => {
+        if (lastFocusedElementOutsideTrap.current) {
+          UIManager.focus(lastFocusedElementOutsideTrap.current);
```
Probably need to check if this element is still in the document

necolas reviewed (changes requested):
We will need unit tests for this behavior

RafikiTiki commented:
@necolas sorry that it did take so long but I've finally found some time to update the PR. I've refactored things you've pointed out and also implemented tests for refocusing mechanism. Have a good day :)

necolas commented:
Thanks!
